app/src/main/java/org/simple/clinic/facility/change/FacilityChangeScreenController.kt

```diff
@@ -96,6 +98,23 @@ class FacilityChangeScreenController @Inject constructor(
     }
   }
 
+  private fun showProgressForReadingLocation(events: Observable<UiEvent>): Observable<UiChange> {
+    val screenCreations = events.ofType<ScreenCreated>()
+    val locationPermissionChanges = events
+        .ofType<FacilityChangeLocationPermissionChanged>()
+        .map { it.result }
+
+    val showProgress = Observables.zip(screenCreations, locationPermissionChanges)
+        .filter { (_, permissionResult) -> permissionResult == GRANTED }
```

Comment:
Can't the `filter` happen before we `zip` i.e. in `locationPermissionChanges`?

Reply:
Also, that will mean that `showProgress` will always happen if the stream is active.
Does it make sense?

Reply:
No, the `filter` should happen after zipping or else we can see stale values. However it does not matter in the current situation because both streams will only emit once. We aren't observing changes to location permissions and only picking up the latest value.

Reply:
If that is the case, then please consider using `withLatestFrom` instead.

Reply:
Discussed externally: I've updated the commit to use `combineLatest()` along with `take(1)` so it's easier to understand.